IRremoteESP8266.cpp

```diff
@@ -702,23 +702,37 @@ static void ICACHE_RAM_ATTR read_timeout(void *arg __attribute__((unused))) {
 
 static void ICACHE_RAM_ATTR gpio_intr() {
   uint32_t gpio_status = GPIO_REG_READ(GPIO_STATUS_ADDRESS);
+  uint32_t now = system_get_time();
+  static uint32_t start = 0;
+
+  os_timer_disarm(&timer);
   GPIO_REG_WRITE(GPIO_STATUS_W1TC_ADDRESS, gpio_status);
 
-	if (irparams.rcvstate == STATE_STOP) {
-		return;
-	}
-  static uint32_t start = 0;
-  uint32_t now = system_get_time();
-	if (irparams.rcvstate == STATE_IDLE) {
-		irparams.rcvstate = STATE_MARK;
-		irparams.rawbuf[irparams.rawlen++] = 20;
-	} else if (irparams.rawlen < RAWBUF) {
-		irparams.rawbuf[irparams.rawlen++] = (now - start) / USECPERTICK + 1;
+  if (irparams.rawlen >= RAWBUF) {
+    irparams.overflow = true;
+    irparams.rcvstate = STATE_STOP;
   }
-  start = now;
 
-  os_timer_disarm(&timer);
-  os_timer_arm(&timer, 15, 0);
+  if (irparams.rcvstate == STATE_STOP) {
+    return;
+  }
+
+  if (irparams.rcvstate == STATE_IDLE) {
+    irparams.overflow = false;
+    irparams.rcvstate = STATE_MARK;
+    irparams.rawbuf[irparams.rawlen++] = 1;
+  } else {
+    uint32_t passed;
+    if (now < start)
+      passed = 0xFFFFFFFF - start + now;
+    else
+      passed = now - start;
```

Comment:
Wow, really? at 80MHz I would have thought this sort of thing should well sub-uSecond.